Mathlib/Topology/Algebra/InfiniteSum/UniformOn.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2025 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import Mathlib.Analysis.Asymptotics.Lemmas
+import Mathlib.Topology.Algebra.UniformConvergence
+
+
+/-!
+# Infinite sum and products that converge uniformly on a set
+
+This file defines the notion of uniform convergence of infinite sums and products on a set
+`s : Set β`. It also defines the notion of local uniform convergence of infinite sums and products
+on a set.
+
+-/
+
+noncomputable section
+
+open Filter Function
+
+open scoped Topology
+
+variable {α β ι : Type*}
+
+section HasProdUniformlyOn
+
+variable [CommMonoid α] {f : ι → β → α} {g : β → α} {s : Set β}
+
+@[simp]
+lemma ofFun_prod (f : ι → β → α) (g : β → α) (s : Set β) (i : Finset ι) :
+    ∏ b ∈ i, (UniformOnFun.ofFun {s}) (f b) = (UniformOnFun.ofFun {s}) (∏ b ∈ i, f b) := rfl
+
+variable [UniformSpace α]
+
+/-- `HasProdUniformlyOn f g` means that the (potentially infinite) product of the `f b` for `b : β`
+converges uniformly on `s` to `g`. -/
+@[to_additive]
+def HasProdUniformlyOn (f : ι → β → α) (g : β → α) (s : Set β) : Prop :=
+  HasProd (fun i ↦ UniformOnFun.ofFun {s} (f i)) (UniformOnFun.ofFun {s} g)
```

Comment:
It is more general, and no more difficult, to allow a general set of sets (mirroring the existing "TendstoUniformlyOn" API), rather than just `{s}`. Then you can do e.g. `HasProdUniformlyOn f g (Compacts β)` or something like that, for uniform convergence on compacts (which should be equivalent to locally-uniform convergence when `β` is locally compact).

Reply:
Unfortunately, `Compacts` has the wrong type for this. We could make it manually but maybe we dont bother.